cps/kni.c

```diff
@@ -850,6 +850,273 @@ rd_send_err(const struct nlmsghdr *req, int err)
 	}
 }
 
+static void
+rd_fill_getroute_reply(struct cps_config *cps_conf, struct nlmsghdr *reply,
+	struct gk_fib *fib, int family, uint32_t seq,
+	uint8_t prefix_len, struct ipaddr **gw_addr)
+{
+	struct rtmsg *rm;
+
+	reply->nlmsg_type = RTM_NEWROUTE;
+	reply->nlmsg_flags = NLM_F_MULTI;
+	reply->nlmsg_seq = seq;
+	reply->nlmsg_pid = cps_conf->nl_pid;
+
+	rm = mnl_nlmsg_put_extra_header(reply, sizeof(*rm));
+	rm->rtm_family = family;
+	rm->rtm_dst_len = prefix_len;
+	rm->rtm_src_len = 0;
+	rm->rtm_tos = 0;
+	rm->rtm_table = RT_TABLE_MAIN;
+	rm->rtm_scope = RT_SCOPE_UNIVERSE;
+	rm->rtm_type = RTN_UNICAST;
+	rm->rtm_flags = 0;
+
+	switch (fib->action) {
+	case GK_FWD_GRANTOR:
+		mnl_attr_put_u32(reply, RTA_OIF, cps_conf->back_kni_index);
+		rm->rtm_protocol = RTPROT_STATIC;
+		*gw_addr = &fib->u.grantor.eth_cache->ip_addr;
+		break;
+	case GK_FWD_GATEWAY_FRONT_NET:
+		mnl_attr_put_u32(reply, RTA_OIF, cps_conf->front_kni_index);
+		rm->rtm_protocol = fib->u.gateway.rt_proto;
+		*gw_addr = &fib->u.gateway.eth_cache->ip_addr;
+		break;
+	case GK_FWD_GATEWAY_BACK_NET:
+		mnl_attr_put_u32(reply, RTA_OIF, cps_conf->back_kni_index);
+		rm->rtm_protocol = fib->u.gateway.rt_proto;
+		*gw_addr = &fib->u.gateway.eth_cache->ip_addr;
+		break;
+	case GK_FWD_NEIGHBOR_FRONT_NET:
+		mnl_attr_put_u32(reply, RTA_OIF, cps_conf->front_kni_index);
+		rm->rtm_protocol = RTPROT_UNSPEC;
```

Comment:
Shouldn't we use`RTPROT_STATIC` here and in the `case GK_FWD_NEIGHBOR_BACK_NET`?

Reply:
Can we assume that it will be `RTPROT_STATIC` because of this comment?

https://github.com/AltraMayor/gatekeeper/blob/master/gk/fib.c#L1094-L1100

And if so, is the following comment incorrect?

https://github.com/AltraMayor/gatekeeper/blob/master/gk/fib.c#L1555-L1556

Reply:
I understand `RTPROT_UNSPEC` as coming from an unknown source, and `RTPROT_STATIC` as coming from a fix configuration (or manually entered).

The first comment is just pointing out a code limitation. And reinforcing that the neighbor prefix is fixed during boot.

The second comment is not precisely written, but I understood it as an indirect reference to the first comment.

Reply:
OK, so it seems that for now we can assume `RTPROT_STATIC`, but if this code limitation was removed we could track the protocol of neighbor entries like we do with other entries.

Reply:
Correct.